edx/commerce-coordinator
# feat: added a management command to start celery with auto-reload

**Description:** Added a management command to start a celery worker process with Django's autoreload watchdog, so the worker will be restarted whenever the code changes; this should make local development easier and was suggested in [PR #7](https://github.com/edx/commerce-coordinator/pull/7)

**JIRA:**[[REV-2484] [ER-Impl] Set up Celery for auto-reload - JIRA](https://openedx.atlassian.net/browse/REV-2484)

**Testing instructions:**

- Set up and activate a virtualenv as described in [the README](https://github.com/edx/commerce-coordinator#one-time-setup) and [PR #7](https://github.com/edx/commerce-coordinator/pull/7)
- Map the redis port and launch it in devstack as described in [PR #7](https://github.com/edx/commerce-coordinator/pull/7) "Local Testing With Celery"
- Start commerce-coordinator with `./manage.py runserver localhost:8000 --settings=commerce_coordinator.settings.local` in one terminal
- Start the worker with `./manage.py celery` in another terminal
- In a third terminal, run `curl -s "http://localhost:8000/lms/test/" | jq '.'`
- Make a code change and save it; you should see celery restart automatically in the second terminal

**Author concerns:** I'm not sure any of our other projects run celery this way; perhaps this will change once we productionalize the coordinator into docker/devstack
